### PR TITLE
basic color preferences and color handling changed to avoid copying

### LINKED ISSUE #16: Compare colors customization
Just reminder for myself to make that somewhen

### PR DESCRIPTION
**Pull request by pvanek**
QColor objects in favor of passing enums

It should resolve #16  and #23

### CODE REVIEW

File: preferences/preferencesdialog.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    delete ui;
+}
+
+void PreferencesDialog::setupButton(QPushButton *b, const QColor &c)
+{
+    QPixmap pm(64, 64);
+    pm.fill(c);
+    QIcon icon(pm);
+    b->setIcon(icon);
```
Comment (coozoo) [inline preferences/preferencesdialog.cpp:54]:
```suggestion
    b->setIcon(icon);
    b->setStyleSheet("text-align:left;");
```

Comment (coozoo) [reply preferences/preferencesdialog.cpp:54]: In such case window looks more in order not like a colored chess board :)
Sure maybe you have another ideas how to align them better

**Comment by coozoo** [comment]
When I first looked on code really I don't liked Idea to include preferences into model. But after going all changes I think it really great idea and work :) thank you.
The only what a bit strage again related to settings window I think maybe it's good to add some reset button If you messed up with color settings and want to reset them :)
And maybe it's good to mark bottom group with label "Text view:" or something like that?

File: preferences/preferencesdialog.cpp (hunk 1 above)
Comment (pvanek) [reply preferences/preferencesdialog.cpp:54]: Text align QSS won't work in RTL language setups. What about to change the dialog to something like this?

Screenshot_20201219_143724

**Comment by pvanek** [comment]
yes, the "restore defaults" is a good idea. I'll do it.

File: preferences/preferencesdialog.cpp (hunk 1 above)
Comment (coozoo) [reply preferences/preferencesdialog.cpp:57]: I think it's perfect :)

**Pull request merged**